var/spack/repos/builtin/packages/tensorflow/package.py

```diff
@@ -0,0 +1,236 @@
+                        '.tf_configure.bazelrc')
+        if spec.satisfies('@1.6.0:'):
+            # tensorboard name changed
+            filter_file(r"'tensorboard >=",
+                        r"#'tensorboard >=",
+                        'tensorflow/tools/pip_package/setup.py')
+        if spec.satisfies('@1.8.0:'):
+            # 1.8.0 and 1.9.0 aborts with numpy import error during python_api generation
+            # somehow the wrong PYTHONPATH is used...set --distinct_host_configuration=false as a workaround
+            # (https://github.com/tensorflow/tensorflow/issues/22395#issuecomment-431229451)
+            filter_file('build --action_env TF_NEED_OPENCL_SYCL="0"',
+                        'build --action_env TF_NEED_OPENCL_SYCL="0"\n'
+                        'build --distinct_host_configuration=false\n'
+                        'build --action_env PYTHONPATH="{}"'.format(env['PYTHONPATH']),
+                        '.tf_configure.bazelrc')
+
+        if spec.satisfies('@1.12.0:'):
+            # add link to spack-installed openssl libs (needed if no system openssl available)
+            filter_file('-lssl', '-lssl '+spec['openssl'].libs.search_flags, 'third_party/systemlibs/boringssl.BUILD')
+            filter_file('-lcrypto', '-lcrypto '+spec['openssl'].libs.search_flags, 'third_party/systemlibs/boringssl.BUILD')
+
+        if '+cuda' in spec:
+            # get path for all dependent libraries to avoid issue with linking especially cuda and cudnn
+            ld_lib_path = env.get('LD_LIBRARY_PATH')
+            bazel('-c', 'opt', '--config=cuda', '--action_env="LD_LIBRARY_PATH=%s"' % ld_lib_path,  '//tensorflow/tools/pip_package:build_pip_package')
+        else:
+            bazel('-c', 'opt', '--config=mkl', '//tensorflow/tools/pip_package:build_pip_package')
+
+        build_pip_package = Executable('bazel-bin/tensorflow/tools/pip_package/build_pip_package')
+        build_pip_package('.')
+
+        # using setup.py for installation
+        # webpage suggests: sudo pip install /tmp/tensorflow_pkg/tensorflow-0.XYZ.whl
+        mkdirp('_python_build')
+        cd('_python_build')
+        ln = which('ln')
+
+        for fn in glob("../bazel-bin/tensorflow/tools/pip_package/build_pip_package.runfiles/org_tensorflow/*"):
+            ln('-s', fn, '.')
+        for fn in glob("../tensorflow/tools/pip_package/*"):
+            ln('-s', fn, '.')
+        setup_py('install', '--prefix={0}'.format(prefix))
+
+
+    def setup_environment(self, spack_env, run_env):
+        spec = self.spec
+
+        # add libcuda.so to LD_LIBRARY_PATH as build can be triggered on the node without cuda driver
+        #if '+cuda' in spec:
+        #    spack_env.prepend_path('LD_LIBRARY_PATH', spec['cuda'].prefix.lib64.stubs)
+
+        # Note : do not enable skylake yet, see https://github.com/easybuilders/easybuild-easyconfigs/issues/5936
+        opt_flags = "--copt=-mavx --copt=-mavx2 --copt=-mfma --copt=-msse4.2 --copt=-mfpmath=both"
```

Comment:
Do we want to have these unconditionally?

Reply:
as target is selected at runtime, it's ok to add these targets unconditionally here.